### PR TITLE
A mock FA2 contract

### PR DESCRIPTION
**Pull request by gkaracha**
We'll need this one for testing that both collateral configurations (TEZ or FA2) work as expected. Merging it can certainly wait.

### CODE REVIEW

**Review by dorranh — APPROVED** [review]
This also looks good to me :slightly_smiling_face:

**Comment by gkaracha** [comment]
> This also looks good to me slightly_smiling_face

Hooray! I'll just wait for CI on the `master` branch to finish before merging this one.

**Pull request merged**
